core/src/worker/mod.rs

```diff
@@ -352,7 +365,17 @@ impl Worker {
             info!("Activity polling is disabled for this worker");
         };
         let la_sink = LAReqSink::new(local_act_mgr.clone(), config.wf_state_inputs.clone());
+        let uuid = uuid::Uuid::new_v4().simple().to_string();
+        let provider = SlotProvider::new(
+            uuid.clone(),
+            config.namespace.clone(),
+            config.task_queue.clone(),
+            wft_semaphore.clone(),
+            external_wft_tx,
+        );
+        client.workers().write().register(Box::new(provider));
         Self {
```

Comment:
Hrmm, wonder if it's clearer for the worker to just impl `SlotProvider` trait directly

Reply:
At first I started that way, but I had to Arc workers + Weak refs, and the comparisons with the hashmap+vector were getting unnecessary ugly, and it required to do more intrusive changes because Workers::new() don't  return Arc workers refs.  
It was also more difficult to debug compared to just using simple uuid strings, so I think the indirection helps...